molecule/source/prepare.yml

```diff
@@ -23,7 +23,7 @@
 
     - name: Clone pulp plugin repositories
       git:
-        repo: 'https://github.com/pulp/{{ item.key| replace("-", "_") }}.git'
+        repo: '{{ item.value.git_repo | default(https://github.com/pulp/{{ item.key| replace("-", "_") }}.git) }}'
```

Comment:
This double nested curley brackets, and no quotes around the https string, might not work. If it doesn't, try:
repo: '{{ item.value.git_repo | default( "https://github.com/pulp/" + item.key | replace("-", "_") ) }}'